packages/components/app/styles/components/application-state.scss

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APPLICATION-STATE COMPONENT
+//
+$hds-application-state-padding: 12px 0;
+
+.hds-application-state {
+  align-self: center;
+  width: 20rem;
+  max-width: 100%;
+  margin: 0 auto;
+}
+
+.hds-application-state__header,
+.hds-application-state__body,
+.hds-application-state__footer {
+  display: flex;
+}
+
+.hds-application-state__header {
+  display: grid;
+  grid-row-gap: 0;
+  grid-column-gap: 0;
+  color: var(--token-color-foreground-primary);
+
+  .hds-application-state--error & {
+    grid-template-rows: repeat(2, auto);
+    grid-template-columns: 28px auto;
+    align-items: stretch;
+
+    .hds-application-state__header-icon {
+      grid-area: 1 / 1 / 3 / 2;
+      padding-top: 4px;
```

Comment:
You may want to expose the padding-top value as a token. (In case the padding doesn't look quite right with some icons. Another option could be to restructure the layout to center align the icon to the "hds-application-state__header-title" text but that would require some involved re-working.)